weave

```diff
@@ -319,6 +320,13 @@ elif [ "$1" = "run" -a -z "$IS_LOCAL" ] ; then
     exit 0
 fi
 
+machineid() {
+    [ -e "/etc/machine-id" ] && cat "/etc/machine-id" && return 0 || true
+    [ -e "/var/lib/dbus/machine-id" ] && cat "/var/lib/dbus/machine-id" && return 0 || true
+}
+
+[ -n "$IS_LOCAL" ] || MACHINE_ID=$(machineid)
```

Comment:
We are not interested in the docker client's machine id but the server's. So the machine id must be determined on the server.